internal/payloads/error_payload.go

```diff
@@ -66,40 +69,89 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
 	}
 }
 
-func ClientError(ctx context.Context, client string, message string, err error, status int) *ResponseError {
-	msg := fmt.Sprintf("%v error: %s: %v", client, message, err)
+func ClientErrorHelper(err error) (int, string) {
+	if errors.Is(err, clients.NotFoundErr) {
+		return 404, "service returned not found or no data"
+	} else if errors.Is(err, clients.UnauthorizedErr) {
+		return 401, "service returned unauthorized"
+	} else if errors.Is(err, clients.ForbiddenErr) {
+		return 403, "service returned forbidden"
+	} else if errors.Is(err, clients.Non2xxResponseErr) {
+		return 500, "service did not return 2xx"
+	}
+	return 0, ""
```

Comment:
I'd love to see this being `500, "Unknown client error"` and logging it as warning into logs:
`logger.Warn().Err(err).Msg("Unrecognized client error")`

This would account for situation we add a new error in clients but forget to add it here.

Reply:
Actually you cannot do that, look at the call site of these three functions - it tests for zero and in that case there is a final `else` 500 statement.

It is clunky, I agree. I would probably keep having three separate `NewSourceError`, `NewImageBuilderError` and `NewHttpClientError` function doing this separately. That is sort of cleaner and we keep the context. But I am also fine with this "one function rules them all" solution which has big advantage if this is successful - we can simply start using just a single error payload function if we apply the same technique everywhere.

Reply:
Oh, right, I missed that final case, that make sense :+1:
Lets do it then, I'm good :+1: